modules/arm_plugin/demos/coco_detection_android_demo/app/src/main/java/org/intel/openvino/MainActivity.java

```diff
@@ -0,0 +1,273 @@
+                    int length = inputStream.read(buffer);
+                    while (length > 0) {
+                        outputStream.write(buffer, 0, length);
+                        length = inputStream.read(buffer);
+                    }
+                    outputStream.flush();
+                    outputStream.close();
+                    inputStream.close();
+                } catch (Exception e) {
+                    Log.e("CopyError", "Copying model has failed.");
+                    System.exit(1);
+                }
+            }
+        }
+    }
+
+    private void createRandomColor(int length) {
+        Random r = new Random(999);
+        randomColor = new Scalar[length];
+
+        for (int i = 0; i < length; i++) {
+            randomColor[i] = new Scalar(r.nextDouble() * 255, r.nextDouble() * 255, r.nextDouble() * 255);
+        }
+    }
+
+    private void processNetwork() {
+        // Set up camera listener
+        mOpenCvCameraView = (CameraBridgeViewBase) findViewById(R.id.CameraView);
+        mOpenCvCameraView.setVisibility(CameraBridgeViewBase.VISIBLE);
+        mOpenCvCameraView.setCvCameraViewListener(this);
+        mOpenCvCameraView.setCameraPermissionGranted();
+        mOpenCvCameraView.enableFpsMeter();
+        mOpenCvCameraView.setMaxFrameSize(320, 240);
+
+        // Initialize model
+        copyFiles();
+        Core core = new Core(modelDir + "/" + PLUGINS_XML);
+        Model net = core.read_model(modelDir + "/" + MODEL_XML);
+        System.out.println("load ok...");
+
+        // Set config of the network
+        PrePostProcessor p = new PrePostProcessor(net);
+        p.input()
+                .tensor()
+                .set_element_type(ElementType.u8)
+                .set_layout(new Layout("NHWC"));
+
+        p.input().preprocess().resize(ResizeAlgorithm.RESIZE_LINEAR);
+        p.input().model().set_layout(new Layout("NCHW"));
+        p.build();
+
+        CompiledModel compiledModel = core.compile_model(net, DEVICE_NAME);
+        inferRequest = compiledModel.create_infer_request();
+
+        // System info
+        String tag = "APPActivity";
+        ActivityManager activityManager = (ActivityManager) getSystemService(ACTIVITY_SERVICE);
+        ActivityManager.MemoryInfo info = new ActivityManager.MemoryInfo();
+        activityManager.getMemoryInfo(info);
+        Log.i(tag, "residue memory : " + (info.availMem >> 20) + "M");
+
+        tm = new TickMeter();
+        createRandomColor(COCO_CLASSES_91.length);
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        try{
+            System.loadLibrary(OPENCV_LIBRARY_NAME);
+            System.loadLibrary(Core.NATIVE_LIBRARY_NAME);
+        } catch (UnsatisfiedLinkError e) {
+            Log.e("UnsatisfiedLinkError",
+                    "Failed to load native OpenVINO libraries\n" + e.toString());
```

Comment:
In case of an error, you need to know which exactly library could not be loaded, so maybe:
```
Log.e("UnsatisfiedLinkError",
                    "Failed to load native OpenCV library\n" + e.toString());
```
and `Core.loadNativeLibs()` to load OpenVINO

Reply:
Thanks! :)